htlcswitch/link_test.go

```diff
@@ -1494,8 +1494,8 @@ func newSingleLinkTestHarness(chanAmt, chanReserve btcutil.Amount) (
 
 	const startingHeight = 100
 	aliceLink := NewChannelLink(aliceCfg, aliceChannel, startingHeight)
-	if err := aliceSwitch.AddLink(aliceLink); err != nil {
-		return nil, nil, nil, nil, nil, err
+	start := func() error {
```

Comment:
Why's this necessary?

Reply:
This is done such that we can later set up the link to fail when it is added to the switch. (in this case we use it to make `syncStates` fail at startup)